cli/pa_analyze_command.go

```diff
@@ -0,0 +1,980 @@
+			return Skipped, fmt.Errorf("failed to load variables for server %s: %w", serverTag.Value, err)
+		}
+
+		version := serverVarz.Version
+
+		_, exists := versionsToServersMap[version]
+		if !exists {
+			versionsToServersMap[version] = []string{}
+			versionsList = append(versionsList, version)
+			examples.Addf("%s - %s", serverName, version)
+		}
+		versionsToServersMap[version] = append(versionsToServersMap[version], serverName)
+	}
+
+	if len(versionsList) == 1 {
+		cmd.logInfo("All servers are running version %s", versionsList[0])
+	} else {
+		cmd.logIssue("Servers are running %d different versions", len(versionsList))
+		cmd.logExamples(examples)
+	}
+
+	return outcome, nil
+}
+
+func (cmd *paAnalyzeCmd) checkServerHealth(r *archive.Reader) (checkStatus, error) {
+	serverTags := r.ListServerTags()
+	artifactType := archive.TagHealth()
+	notHealthy, healthy := 0, 0
+	examples := newCollectionOfExamples(cmd.examplesLimit)
+
+	for _, serverTag := range serverTags {
+		serverName := serverTag.Value
+		var health server.HealthStatus
+
+		err := r.Load(&health, &serverTag, artifactType)
+		if errors.Is(err, archive.ErrNoMatches) {
+			cmd.logWarning("Artifact 'HEALTHZ' is missing for server %s", serverName)
+			continue
+		} else if err != nil {
+			return Skipped, fmt.Errorf("failed to load health for server %s: %w", serverName, err)
+		}
+
+		if health.Status != "ok" {
+			examples.Addf("%s: %d - %s", serverName, health.StatusCode, health.Status)
+			notHealthy += 1
+		} else {
+			healthy += 1
+		}
+	}
+
+	if notHealthy > 0 {
+		cmd.logIssue("%d/%d servers are not healthy", notHealthy, healthy+notHealthy)
+		cmd.logExamples(examples)
+		return SomeIssues, nil
+	}
+
+	cmd.logInfo("%d/%d servers are healthy", healthy, healthy)
+	return Pass, nil
+}
+
+// checkSlowConsumers alerts for any slow consumer found on any server
+func (cmd *paAnalyzeCmd) checkSlowConsumers(r *archive.Reader) (checkStatus, error) {
+	serverTags := r.ListServerTags()
+	totalSlowConsumers := int64(0)
+	serversWithSlowConsumers := make(map[string]int64)
+	examples := newCollectionOfExamples(cmd.examplesLimit)
+
+	for _, serverTag := range serverTags {
+		serverName := serverTag.Value
+		var serverVarz server.Varz
+		err := r.Load(&serverVarz, &serverTag, archive.TagServerVars())
+		if err != nil {
+			return Skipped, fmt.Errorf("failed to load Varz for server %s: %w", serverName, err)
+		}
+
+		if slowConsumers := serverVarz.SlowConsumers; slowConsumers > 0 {
+			serversWithSlowConsumers[serverName] = slowConsumers
+			examples.Addf("%s: %d slow consumers", serverName, slowConsumers)
+			totalSlowConsumers += slowConsumers
+		}
+	}
+
+	if totalSlowConsumers > 0 {
+		cmd.logIssue("Total slow consumers: %d over %d servers", totalSlowConsumers, len(serversWithSlowConsumers))
+		cmd.logExamples(examples)
+		return SomeIssues, nil
+	}
+	return Pass, nil
+}
+
+const checkClusterMemoryUsageOutlierThreshold = 0.5 // Warn if one node is using over 1.5x the average
```

Comment:
@jarretlavallee

Reply:
Looks reasonable to me. It may give some false positives when the memory footprint is sub 1GiB where a spike could cause a couple hundred MB of growth. It will be good to have some customer data to validate that.